### PR TITLE
PAT-268 Add check your answers screen with arbitrary banding calculation, rework caselist nominals to use XTACT and existing data, prevent doublE click

### PR DESCRIPTION
**Pull request by yasinmustafa**

### CODE REVIEW

File: server/routes/cimList.js [hunk 1]
```diff
@@ -86,5 +86,28 @@ module.exports = ({ offenderService, nominalService }) => router => {
     })
   )
 
+  router.get(
+    '/checkYourAnswers/:bookingId',
+    asyncMiddleware(async (req, res) => {
+      const { bookingId } = req.params
+      const offenderData = await offenderService.getOffenderDetails(res.locals.user.token, bookingId)
+      const nominalData = await nominalService.getNominalByBookingId(bookingId)
+      res.render('pages/cim/checkYourAnswers', {
```
Comment (timharrison-moj) [inline server/routes/cimList.js]: Not sure of the distinction between nominal service and offender service? Might just be a naming issue?
Nominals = people already flagged for interest.
Offender = wider prison population.

File: server/services/nominalService.js [hunk 2]
```diff
@@ -34,55 +34,70 @@ module.exports = function createNominalService({ nominalClient, elite2ClientBuil
...
     )
 
-    const nominalsWithReleaseDates = joinNominalsWithReleaseDate(nominals, bookingIdsWithReleaseDates)
-    const nominalsWithDisplayName = nominalsWithReleaseDates.map(sourceObject => {
-      const nominal = Object.assign({}, sourceObject)
-      const displayName = `${properCaseName(nominal.lastName)}, ${properCaseName(nominal.firstName)} `
-      nominal.displayName = displayName
-      return nominal
+    const xtactAndNominals = joinByBookingId(allExistingBookings, xtactBookings)
+    const bookingIds = xtactAndNominals.map(nominal => nominal.bookingId)
```
Comment (timharrison-moj) [inline server/services/nominalService.js]: It is probably just naming conventions confusing me here. The two arrays are -  1) all offenders in the current user's caseload with the XTACT set and 2) all offenders currently in the Pathfinder database. Not sure if 2) is visible to all Pathfinder users or only visible where they fall into the current user's caseload? When selecting the joined list to display, I think for a bookingId which exists in both arrays it should probably show the Pathfinder DB version?

File: server/routes/cimList.js (hunk 1 above)
Comment (yasinmustafa) [reply server/routes/cimList.js]: The offender service is a service we inherited from UOF - used to get the the offender details aa summarised at the top of each page and the image. This only runs against elte2Api

The nominal service is used to get specific data for pathfinder which includes queries to elite2api and the pathfinder DB. Maybe a better name would be pathfinderService?

File: server/services/nominalService.js (hunk 2 above)
Comment (yasinmustafa) [reply server/services/nominalService.js]: yes you are right, at the moment it shows all XTACTS that are in the user caseloads + any other bookings that are in PF. This second part needs refining.. would prefer to defer the resolution of this after a refinement session and a future story

File: server/routes/cimList.js (hunk 1 above)
Comment (timharrison-moj) [reply server/routes/cimList.js]: Agreed.

File: server/services/nominalService.js (hunk 2 above)
Comment (timharrison-moj) [reply server/services/nominalService.js]: Agreed

**Pull request merged**